community/io/src/test/java/org/neo4j/graphdb/mockfs/EphemeralFileSystemAbstraction.java

```diff
@@ -1269,11 +1283,24 @@ private void verifySize( int totalAmount )
 
         public void clear()
         {
+            assertNotFreed();
             this.buf.clear();
         }
 
+        private void assertNotFreed()
+        {
+            if ( this.buf == null )
+            {
+                String message = "This buffer have been freed";
+                throw this.freeCall != null
+                        ? new IllegalStateException( message, freeCall )
+                        : new IllegalStateException( message );
```

Comment:
It's perfectly legal for the cause of an exception to be `null`, so this code can be simplified.